### PR TITLE
chore: fix surefire caused build failures

### PR DESCRIPTION
**Pull request by stevenpyzhang**
### Description 
_What behavior do you want to change, why, how does your patch achieve the changes?_

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -155,6 +155,8 @@
         <netty.version>4.1.72.Final</netty.version>
         <netty-codec-http2-version>4.1.72.Final</netty-codec-http2-version>
         <jersey-common>2.34</jersey-common>
+
+        <maven.surefire.version>3.0.0-M5</maven.surefire.version>
```
Comment (jnh5y) [inline pom.xml:159]: Any reason not to use 3.0.0-M6? 

Or is an issue caused by that newer version?

**Review by jnh5y — APPROVED** [review]
My question about using M6 is non-blocking.  :shipit: when ready.

File: pom.xml (hunk 1 above)
Comment (stevenpyzhang) [reply pom.xml:159]: I'm waiting to see if the jenkins build passes with this pin, the confluent commons got updated to M6 which is when build errors started causing. As to why the new version causes build errors, I haven't dug into that yet.

**Comment by mjsax** [comment]
I cherry-picked this fix to `7.2.x` branch that is also affected.

**Pull request merged**